library.c

```diff
@@ -2310,9 +2386,12 @@ PHP_REDIS_API int redis_sock_connect(RedisSock *redis_sock)
                 redis_sock->stream = *(php_stream **)zend_llist_get_last(&p->list);
                 zend_llist_remove_tail(&p->list);
 
-                if (redis_sock_check_liveness(redis_sock) == SUCCESS) {
+                if (redis_stream_detect_dirty(redis_sock->stream) == SUCCESS &&
```

Comment:
Why `redis_stream_detect_dirty` is here and not inside `redis_sock_check_liveness`?